### ISSUE TITLE
0xboriskataa - Routing info is always stored at 0 slot of `lotRouting[]` mapping

### ISSUE THREAD

**Issue by sherlock-admin4** [sherlock-audit/2024-03-axis-finance-judging#46]
0xboriskataa

medium

# Routing info is always stored at 0 slot of `lotRouting[]` mapping

## Summary
When creating an auction lot `lotRouting[lotId]` mapping is used to store the routing information.

## Vulnerability Detail
`lotId` is declared in the `returns()` part of `auction()` and it's value is 0 by default. However it is not updated before the mapping `lotRouting[lotId]` is called. It is updated to `lotId=lotCounter++` only after using it in the mapping.

This means that every time we call `auction()`, `lotId` will be declared again and yet again start at 0 thus we will only be overriding the 0th slot of the `lotRouting[]` mapping.
```solidity
    function auction(
        RoutingParams calldata routing_,
        Auction.AuctionParams calldata params_,
        string calldata infoHash_
    ) external nonReentrant returns (uint96 lotId) { // lotId = 0 as a default value
        ...
        Routing storage routing = lotRouting[lotId]; // loitId not updated so it's still 0
        ...
        lotId = lotCounter++; // lotId updated only after mapping has already been refrenced
```
## Impact
You cannot store the routing information for more than one auction lot because you will always be overriding the first one.

## Code Snippet
https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/bases/Auctioneer.sol#L164

https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/bases/Auctioneer.sol#L174

https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/bases/Auctioneer.sol#L194

## Tool used

Manual Review

## Recommendation
Update `lotId` before using it in the mapping:

```solidity
lotId = lotCounter++;
Routing storage routing = lotRouting[lotId];
```

Duplicate of #12
